### ISSUE TITLE
[WIDGET]  Grid Widget

### ISSUE THREAD

**Issue by rice7th** [elkowar/eww#637]
### Description of the widget

A Grid widget is a widget similar to a `box` widget that lets the user define a custom layout just like a CSS grid or Flexbox.

### Implementation proposal

This is probably the closest thing GTK has to offer: https://docs.gtk.org/gtk3/class.Grid.html

###### Note: I am sill a beginner in yuck! Expect some errors

This creates a simple widget like shown in the concept below:

```
(grid
    :columns 4
    :rows 3
    :column_gap "8px"
    :row_gap "8px"
    (grid-child
        :column 1
        :row 1
        :column_span 3
        :row_span 2
        (box :class "whatever" :style "background-color: #DF3E3E;"))
    (grid-child
        :column 1
        :row 3
        :column_span 4
        :row_span 1
        (box :class "whatever" :style "background-color: #7CBD27;"))
    (grid-child
        :column 4
        :row 1
        :column_span 1
        :row_span 2
        (box :class "whatever" :style "background-color: #32B5C7;")))
```
![concept](https://user-images.githubusercontent.com/93940240/204851924-eef101b3-8251-42fd-94fe-2363c5d42833.png)

### Additional context

This way, making widget dashboards like [this one](https://github.com/adi1090x/widgets/tree/main/eww/dashboard) or [this one](https://github.com/Axarva/dotfiles-2.0) would be way easier and way less messier to do.

**Comment by mantacid** [comment]
I feel like there’s. Way to do this with what’s already been implemented, though I’m likely wrong. I would still love to see this implemented, but in a way that allows one to make a grid that can adapt to different widths, wrapping its elements to fit within a changing width.

**Comment by rice7th** [comment]
yes @mantacid, but what I am suggesting is more similar to a CSS grid: a complex structure with spans and elements that go over multiple rows and columns

**Comment by mantacid** [comment]
You're right, that would be useful.

**Comment by Inparsian** [comment]
Would also love to see this implemented, especially considering that grids [are in the GTK spec](https://docs.gtk.org/gtk4/class.Grid.html)
